File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprof=0
        n=len(prices)
        minval=prices[0]
        
        for i in range (0,n):
            currprof=prices[i]-minval
            maxprof=max(maxprof,currprof)
            minval=min(minval,prices[i])
        
        return maxprof
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit=INT_MIN;
        int minval=prices[0];

        for(int i=0;i<prices.size();i++){
            int currprofit=prices[i]-minval;
            max_profit=max(max_profit,currprofit);
            minval=min(minval,prices[i]);
        }

        return max_profit;
    }
};